zarr/sync.py

```diff
@@ -1,11 +1,19 @@
 import os
 from collections import defaultdict
 from threading import Lock
+from typing import Protocol
 
 import fasteners
 
 
-class ThreadSynchronizer:
+class Synchronizer(Protocol):
+    """Base class for synchronizers."""
+
+    def __getitem__(self, item):
+        ...
```

Comment:
Thoughts welcome on how to make codecov happy with these lines.

Reply:
I couldn't find anything obvious in https://docs.codecov.com/docs/ignoring-paths, seems like it only has per-file granularity. Since we know the decrease in coverage isn't meaningful, I'd be fine ignoring it here. In the future maybe we put protocols in `types.py` (either a module-local one, or a zarr-wide one)  and tell codecov to ignore it.

Reply:
I've updated the code coverage config to ignore lines that are "...".

Reply:
good idea!